src/main/java/frc/trigon/robot/subsystems/ledstrip/LEDStrip.java

```diff
@@ -1,189 +1,95 @@
-        CANDLE.animate(
-                new RgbFadeAnimation(
-                        brightness,
-                        speed,
-                        this.numberOfLEDs,
-                        this.offset
-                ),
-                animationSlot
-        );
+    void rainbow() {
+        for (int led = 0; led < numberOfLEDs; led++) {
+            final int hue = (int) (rainbowFirstPixelHue + (led * 180 / numberOfLEDs) % 180);
+            LEDStripConstants.LED_BUFFER.setHSV(led + indexOffset, hue, 255, 128);
+        }
+        rainbowFirstPixelHue += 3;
+        rainbowFirstPixelHue %= 180;
     }
 
-    void animateSingleFade(Color color, double speed) {
-        CANDLE.animate(
-                new SingleFadeAnimation(
-                        color.getRed(),
-                        color.getGreen(),
-                        color.getBlue(),
-                        0,
-                        speed,
-                        this.numberOfLEDs,
-                        this.offset
-                ),
-                animationSlot
-        );
+    void threeSectionColor(Color firstSectionColor, Color secondSectionColor, Color thirdSectionColor) {
+        final int ledsPerSection = (int) Math.floor(numberOfLEDs / 3.0);
+        setThreeSectionColor(ledsPerSection, firstSectionColor, secondSectionColor, thirdSectionColor);
     }
 
-    void animateTwinkle(Color color, double speed, TwinkleAnimation.TwinklePercent divider) {
-        CANDLE.animate(
-                new TwinkleAnimation(
-                        color.getRed(),
-                        color.getGreen(),
-                        color.getBlue(),
-                        0,
-                        speed,
-                        this.numberOfLEDs,
-                        divider,
-                        this.offset
-                ),
-                animationSlot
-        );
+    private void setThreeSectionColor(int ledsPerSection, Color firstSectionColor, Color secondSectionColor, Color thirdSectionColor) {
+        for (int i = 0; i < ledsPerSection; i++)
+            setLedColors(inverted ? thirdSectionColor : firstSectionColor, i);
+        for (int i = ledsPerSection; i < 2 * ledsPerSection * 2; i++)
+            setLedColors(secondSectionColor, i);
+        for (int i = 2 * ledsPerSection; i < numberOfLEDs; i++)
+            setLedColors(inverted ? firstSectionColor : thirdSectionColor, i);
     }
 
-    void animateStrobe(Color color, double speed) {
-        CANDLE.animate(
-                new StrobeAnimation(
-                        color.getRed(),
-                        color.getGreen(),
-                        color.getBlue(),
-                        0,
-                        speed,
-                        this.numberOfLEDs,
-                        this.offset
-                ),
-                animationSlot
-        );
+    private void setLedColors(Color color, int index) {
```

Comment:
`setLEDColor`. It's only doing one LED and in the rest of the code LED is all uppercase